provider/resources.go

```diff
@@ -337,7 +327,9 @@ func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) erro
 
 	// lastly let's set the sharedCreds and sharedConfig file. If these are not found then let's default to the
 	// locations that AWS cli will store these values.
-	sharedCredentialsFile := stringValue(vars, "sharedCredentialsFile", []string{"AWS_SHARED_CREDENTIALS_FILE"})
+	var sharedCredentialsFile string
+	sharedCredentialsFile = stringValue(vars, "sharedCredentialsFile", []string{})
+	sharedCredentialsFile = stringValue(vars, "sharedCredentialsFiles", []string{"AWS_SHARED_CREDENTIALS_FILE", "AWS_SHARED_CREDENTIALS_FILES"})
```

Comment:
Are we overwriting the same variable here? if so, can we add a comment as to why?

Also - I'm not sure what it would look like to specify multiple different credentials files paths via env var([AWS does not have a multiple paths options that I could find](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)), but even if we're just looking at the config value, it is an array type in the schema, so I'm pretty sure the `stringValue` function would hiccup here.

Reply:
the order of operations are:

* check sharedCredentialsFile
* check sharedCredentialsFiles
* check AWS_SHARED_CREDENTIALS_FILE
* AWS_SHARED_CREDENTIALS_FILES

That can't be achieved right now - so we added an extra check of the first one. They all look for the same details

Reply:
I don't believe we need to make any changes here

Reply:
To the first point - what I'm saying is, declaring the var and then overwriting it twice with lightly different values is confusing enough to read, at least to me, that your explanation above would be helpful to have as a comment.
if I understand correctly, the `stringValue` function will return `""` if the underlying value isn't a string. The underlying value of `sharedCredentialsFiles` is an array type. We would overwrite `sharedCredentialsFile` with an empty string every time it is set as a config var. I don't think that's the desired behavior?

Reply:
The schema is mutually exclusive here - you can’t set both values

if you want me to spell that out in code then I can - I don’t see the need to do it at this point but I definitely do it today

Reply:
@guineveresaenger this is ready for your review again